packages/next/server/render.tsx

```diff
@@ -501,7 +501,8 @@ export async function renderToHTML(
   }
 
   let renderServerComponentData = isServerComponent
-    ? query.__flight__ !== undefined
+    ? query.__flight__ !== undefined ||
+      (isDataReq && process.env.NEXT_RUNTIME === 'edge')
```

Comment:
We also handle the data requests for RSC pages too (although we currently use the `?__flight__` query parameter hack). In the future we will move to the data URL for consistency, as well as adding streaming and `.rsc` format support.